app/models/proposal.rb

```diff
@@ -75,12 +75,20 @@ def self.recommendations(user)
       .where("author_id != ?", user.id)
       .unsuccessful
       .not_followed_by_user(user)
+      .not_archived
+      .not_supported_by_user(user)
   end
 
   def self.not_followed_by_user(user)
     where.not(id: followed_by_user(user).pluck(:id))
   end
 
+  def self.not_supported_by_user(user)
+    @supported_by_user = user.find_voted_items(votable_type: "Proposal")
+    @supported_by_user = @supported_by_user.reject { |p| p.nil? } if @supported_by_user.include?(nil)
```

Comment:
I tried to reproduce the scenario where the `user.find_voted_items(votable_type: "Proposal")` query would return empty/nil results but couldn't (I tried softdeleting as you said a voted Proposal with `.destroy`)

Reply:
With a freshly seeded database, the admin user has 3 votes: 2 for debates and 1 for a proposal, but that proposal is hidden/soft-deleted. The `find_voted_items` method fetches this record (Apparently, it completely ignores the `acts_as_paranoid` behavior since `user.find_voted_items(votable_type: 'Proposal').count` outputs `1`) as `nil`, which returns the following array:

`[nil, Debate#1, Debate#2]`

If we use the aforementioned method only to fetch an user's upvoted proposals, we would receive the following array: `[nil]`

And then, when the `recommendations` method is chained to the `sort_by_recommendations` scope, it will raise an exception since Ruby's `NilClass` doesn't know anything about this, hence we need to **exclude** those `nil` elements to avoid the error.